### ISSUE TITLE
libc: 'connect()' erroneously reports success when used with VFS server

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#2920]
The test case in commit [1] tries to connect to 'genode.org:81', which should fail, but the 'connect()' call does not fail when using the VFS server.

good case: vfs_lxip plugin used locally:

```
[init -> test-libc_connect] socket() returned 4
[init -> test-libc_connect] connect() returned -1
```

bad case: vfs_lxip plugin used via VFS server:

```
[init -> test-libc_connect] socket() returned 4
[init -> vfs] Error: partial write detected 0 vs 17
[init -> test-libc_connect] Error: packet operation=1 failed
[init -> test-libc_connect] connect() returned 0
```

[1] https://github.com/cproc/genode/commit/987132aed38dacc6f1b15b76cf467f7d4b847324

**Comment by ehmry** [comment]
As I recall there was a solution to this by calling sync after writing to the connect, so either that is not implemented or it doesn't work over the FS session.

**Comment by ehmry** [comment]
Fixed on my `issue2920_libc_connect` branch.

**Comment by ehmry** [comment]
LwIP fix at c3fac5d746ffe43836f33408e50974c662052994.
